### PR TITLE
Fix boot issue for Creality E3 Free-runs board

### PR DESCRIPTION
**Pull request by geijt**
### Description

The primary fix is for linker script to use LD_MAX_DATA_SIZE (as some other boards also do) which is automatically set by offset_and_rename.py to upload.maximum_ram_size - 40 (so just a little lower than it already was) because otherwise the screen stays blank because the board doesn't boot (also no serial ouput).
I've also updated Conditionals_adv.h so the LCD_SERIAL_PORT is automatically set to 2 instead of the default 3 (which doesn't exists on the STM32F401) so there's no compiler error like this:

``` C
Marlin\src\HAL\STM32\MarlinSerial.cpp:56:40: error: 'USART3' was not declared in this scope; did you mean 'USART1'?
   56 |     MSerialT MSerial ## ser_num (true, USART ## ser_num, &_rx_complete_irq_ ## ser_num); \
      |                                        ^~~~~
Marlin\src\HAL\STM32\MarlinSerial.cpp:67:3: note: in expansion of macro 'DECLARE_SERIAL_PORT'
   67 |   DECLARE_SERIAL_PORT(3)
      |   ^~~~~~~~~~~~~~~~~~~
*** [.pio\build\STM32F401RE_freeruns\src\src\HAL\STM32\MarlinSerial.cpp.o] Error 1
```
I know I can manually set LCD_SERIAL_PORT in my configuration, but this way we can prevent issues from being created about this.

And finally, I updated stm32f4.ini with the addition of board_build.rename to create unique filenames as required for Creality printers, removed the build flag ENABLE_HWSERIAL3 (as before, the STM32F401 doesn't have a HWSerial3) and shared a few defines in the pins file of this board.

<br/>
Note: 
I'm already running for a few weeks with these changes applied to those from #25636 on v2.1.2.1 without problems, but unfortunately my PR wasn't merged into the source branch of that PR.

### Requirements

The Creality E3 Free-runs (CR4NT220801C10) board with an Ender-3 V2.

### Benefits

The display will **not** remain blank because the firmware cannot boot.

### Configurations

Basic config example:

``` C
#define MOTHERBOARD BOARD_CREALITY_CR4NTXXC10
#define SERIAL_PORT 1
#define X_DRIVER_TYPE  TMC2209
#define Y_DRIVER_TYPE  TMC2209
#define Z_DRIVER_TYPE  TMC2209
#define E0_DRIVER_TYPE  TMC2209
#define ENDSTOP_INTERRUPTS_FEATURE
#define CLASSIC_JERK
#define BLTOUCH
#define EEPROM_SETTINGS
#define SDSUPPORT
#define DWIN_CREALITY_LCD
```

### Related Issues

N/A

### CODE REVIEW

File: Marlin/src/pins/stm32f4/pins_CREALITY_CR4NTXXC10.h [hunk 1]
```diff
@@ -142,21 +142,21 @@
...
-    #define Y_SLAVE_ADDRESS 1
+    #define Y_SLAVE_ADDRESS                   1
   #endif
   #ifndef Z_SLAVE_ADDRESS
-    #define Z_SLAVE_ADDRESS 2
+    #define Z_SLAVE_ADDRESS                   2
   #endif
   #ifndef E0_SLAVE_ADDRESS
-    #define E0_SLAVE_ADDRESS 3
+    #define E0_SLAVE_ADDRESS                  3
```
Comment (classicrocker883) [inline Marlin/src/pins/stm32f4/pins_CREALITY_CR4NTXXC10.h:159]: based on other similar files, perhaps revert this change

**Review by classicrocker883 — CHANGES_REQUESTED** [review]

**Comment by thinkyhead** [comment]
> unfortunately my PR wasn't merged into the source branch of that PR.

If a busy author fails to notice your PR to their branch, one of the admins can grab those changes and bring them to the PR more directly. Just add a comment about it on the PR and we'll _definitely_ see that before the PR gets merged.
